app/services/course/material/zip_download_service.rb

```diff
@@ -1,12 +1,70 @@
 class Course::Material::ZipDownloadService
-  class << self
-    # Downloads the materials and zip them.
-    #
-    # @param [Course::Material::Folder] folder The folder containing the materials.
-    # @param [Array<Course::Material>] materials The materials to be downloaded.
-    # @return [String] The path to the zip file.
-    def download_and_zip(_folder, _materials)
-      # TODO: To implement
+  attr_accessor :materials, :folder, :base_dir
+
+  # @param [Course::Material::Folder] folder The folder containing the materials.
+  # @param [Array<Course::Material>] materials The materials to be downloaded.
+  def initialize(folder, materials)
```

Comment:
So at this point it's very easy to use instance variables to store state. Decide what you need to use as an instance variable (preferably things that do not change) and the rest should remain parameters. Parameters that are used by every method and not modified by any method can be an instance variable.